tests/test_core_network.py

```diff
@@ -2717,24 +2726,154 @@ def test_generate_validation_report_with_correct_schedule(correct_schedule):
+        {}
+    )
+
+
+@pytest.mark.parametrize("value,offending_value", offending_link_attribute_values_and_names)
+def test_values_of_from_node_are_not_flagged_in_validation_report(value, offending_value):
+    n = Network('epsg:27700')
+    n.add_link('1', offending_value, 2, attribs={'length': 1, 'capacity': 1, 'freespeed': 1, "modes": ['car', 'bus']})
+    n.add_link('2', 2, 3, attribs={'length': 2, 'capacity': 1, 'freespeed': 2, "modes": ['car', 'bus']})
+
+    report = n.generate_validation_report()
+
+    assert_semantically_equal(
+        report['graph']['link_attributes'][f'{value}_attributes'],
+        {}
+    )
+
+
+@pytest.mark.parametrize("value,offending_value", offending_link_attribute_values_and_names)
+def test_values_of_to_node_are_not_flagged_in_validation_report(value, offending_value):
+    n = Network('epsg:27700')
+    n.add_link('1', 1, offending_value, attribs={'length': 1, 'capacity': 1, 'freespeed': 1, "modes": ['car', 'bus']})
+    n.add_link('2', 2, 3, attribs={'length': 2, 'capacity': 1, 'freespeed': 2, "modes": ['car', 'bus']})
+
+    report = n.generate_validation_report()
+
+    assert_semantically_equal(
+        report['graph']['link_attributes'][f'{value}_attributes'],
+        {}
+    )
+
+
 def test_zero_value_attributes_show_up_in_validation_report():
     n = Network('epsg:27700')
     n.add_link('1', 1, 2, attribs={'length': 0, 'capacity': 0.0, 'freespeed': '0.0', "modes": ['car', 'bus']})
     n.add_link('2', 2, 3, attribs={'length': 2, 'capacity': 1, 'freespeed': 2, "modes": ['car', 'bus']})
 
     report = n.generate_validation_report()
-    correct_report = {'graph': {
-        'graph_connectivity': {
-            'car': {'problem_nodes': {'dead_ends': [3], 'unreachable_node': [1]}, 'number_of_connected_subgraphs': 3},
-            'walk': {'problem_nodes': {'dead_ends': [], 'unreachable_node': []}, 'number_of_connected_subgraphs': 0},
-            'bike': {'problem_nodes': {'dead_ends': [], 'unreachable_node': []}, 'number_of_connected_subgraphs': 0}},
-        'link_attributes': {
-            'links_over_1km_length': {'number_of': 0, 'percentage': 0.0, 'link_ids': []},
-            'zero_attributes': {
-                'length': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']},
-                'capacity': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']},
-                'freespeed': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']}}}}}
-    assert_semantically_equal(report, correct_report)
+
+    assert_semantically_equal(
+        report['graph']['link_attributes']['zero_attributes'],
+        {
+            'length': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']},
+            'capacity': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']},
+            'freespeed': {'number_of': 1, 'percentage': 0.5, 'link_ids': ['1']}
+        }
+    )
+
+
+def test_negative_value_attributes_show_up_in_validation_report():
+    n = Network('epsg:27700')
```

Comment:
These tests are very good 👏 

The assertions are nice and narrow, **and** I can see clearly, without leaving the test, _why_ we expect what we expect in those assertions